src/components/modals/LogsModal.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react'
+import { Platform, View, ViewStyle } from 'react-native'
+import { AirshipBridge } from 'react-native-airship'
+import RNFS from 'react-native-fs'
+import Share, { ShareOptions } from 'react-native-share'
+
+import { saveLogOutput } from '../../actions/LogActions'
+import s from '../../locales/strings'
+import { useDispatch } from '../../types/reactRedux'
+import { WarningCard } from '../cards/WarningCard'
+import { CrossFade } from '../common/CrossFade'
+import { showError } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { MainButton } from '../themed/MainButton'
+import { ModalCloseArrow, ModalMessage, ModalTitle } from '../themed/ModalParts'
+import { OutlinedTextInput } from '../themed/OutlinedTextInput'
+import { ThemedModal } from '../themed/ThemedModal'
+interface LogsModalProps {
+  bridge: AirshipBridge<void>
+}
+
+export const LogsModal = ({ bridge }: LogsModalProps) => {
+  const theme = useTheme()
+  const dispatch = useDispatch()
+  const styles = getStyles(theme)
+  const [spinning, setSpinning] = React.useState(false)
+  const [userMessage, setUserMessage] = React.useState('')
+  const [safeToSend, setSafeToSend] = React.useState(true)
```

Comment:
Optional idea: We could call `saveLogOutput` in a `useAsyncEffect` as soon as the modal loads. Then we do the safety check & update the UI to either un-grey the buttons or show the error message. That way you know right away whether the logs are good or not, and can provide that feedback immediately.

Reply:
Going ahead w/ this approach.

Reply:
MultiLogOutput now gets passed as a prop to LogsModal and safety-checked inside the modal, providing immediate feedback.